perception/multi_object_tracker/src/tracker/model/tracker_base.cpp

```diff
@@ -54,6 +54,60 @@ bool Tracker::updateWithoutMeasurement()
   return true;
 }
 
+void Tracker::updateClassification(
+  const std::vector<autoware_auto_perception_msgs::msg::ObjectClassification> & classification)
+{
+  // classification algorithm:
+  // 0. Remove the class with probability < 0.005
+  // 1. Decay all existing probability
+  // 2. Update the matched classification probability with a gain
+  // 3. If the label is not found, add it to the classification list
+  // 4. Normalize
+
+  // Gain and decay
+  const double gain = 0.05;
+  const double decay = 1.0 - gain;
+
+  // If the probability is less than 0.005, remove the class
+  classification_.erase(
+    std::remove_if(
+      classification_.begin(), classification_.end(),
+      [](const auto & class_) { return class_.probability < 0.005; }),
+    classification_.end());
+
+  // Decay all existing probability
+  for (auto & class_ : classification_) {
+    class_.probability *= decay;
+  }
+
+  // Update the matched classification probability with a gain
+  for (const auto & new_class : classification) {
+    bool found = false;
+    for (auto & old_class : classification_) {
+      if (new_class.label == old_class.label) {
+        old_class.probability += new_class.probability * gain;
+        found = true;
+        break;
+      }
+    }
+    // If the label is not found, add it to the classification list
+    if (!found) {
+      auto adding_class = new_class;
+      adding_class.probability *= gain;
+      classification_.push_back(adding_class);
+    }
+  }
+
+  // Normalize
```

Comment:
[imo]
Class normalization may used in input, so I think we can make this part to be a function. (lambda is OK)